powerline/bindings/lemonbar/powerline-lemonbar-multimonitor.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+# vim:fileencoding=utf-8:noet
+from __future__ import (unicode_literals, division, absolute_import, print_function)
+import sys
+import time
+import re
+import subprocess
+
+from threading import Lock, Timer
+from argparse import ArgumentParser, REMAINDER
+
+from powerline.lemonbar import LemonbarPowerline
+from powerline.lib.monotonic import monotonic
+from powerline.lib.dict import mergedicts
+
+
+if __name__ == '__main__':
+	parser = ArgumentParser(
+		description='Powerline BAR bindings.',
+		usage='%(prog)s [-h] [--i3] [--height HEIGHT] [--theme THEME [THEME ...]] -- args')
+	parser.add_argument(
+		'--i3', action='store_true',
+		help='Subscribe for i3 events.'
+	)
+	parser.add_argument(
+		'--height', default='',
+		help='Bar height'
+	)
+	parser.add_argument(
+		'--theme', '-t',
+		default=[], nargs='+',
+		help='makes all outputs that containing a string use the specified theme. Format like "out=theme"'
+	)
+	parser.add_argument(
+		'args', nargs=REMAINDER,
+		help='Extra arguments for lemonbar'
+	)
+	args = parser.parse_args()
+	theme_rules = {}
+	for theme in args.theme:
+		out, theme = theme.split('=', 1)
+		theme_rules[out] = theme
+
+	active_screens = [match.groupdict() for match in re.finditer(
+		'^(?P<name>\w+) connected (?P<width>\d+)x(?P<height>\d+)\+(?P<x>\d+)\+(?P<y>\d+)',
+		subprocess.check_output(['xrandr', '-q']).decode(),
+		re.MULTILINE
+	)]
+
+	powerlines = []
+
+	for screen in active_screens:
+		theme = None
+		for out in theme_rules:
+			if out in screen['name']:
+				theme = theme_rules[out]
+
+		line = LemonbarPowerline(theme=theme, output=screen['name'])
+		command = ['bar-aint-recursive', '-g', '{}x{}+{}'.format(screen['width'], args.height, screen['x'])] + args.args[1:]
+		process = subprocess.Popen(command, stdin=subprocess.PIPE)
+		powerlines.append((line, process, int(screen['width']) / 5))
+
+	lock = Lock()
+	modes = ['default']
+
+	def render(reschedule=False):
+		if reschedule:
+			Timer(0.5, render, kwargs={'reschedule': True}).start()
```

Comment:
there is also no preexisting facility for per-monitor theme configuration. The reason there is a `--theme` in the non-multimonitor version is the same, if you have multiple monitors an override is the only way you can have multiple themes in the same extension at the same time, and that was definetely needed for the workspace segment prior to the `segment_info['output']`. In the `powerline-lemonbar.py` the `--theme` _could_ be removed now because the use-case it was intended for is covered by `…-multimonitor.py` but I am not sure whether there are other reasons someone may need it.